src/ansi.rs

```diff
@@ -715,6 +718,53 @@ bitflags! {
     }
 }
 
+/// Construct escape sequences for regular keys, like `1`, when
+/// modified by `Shift-`, `Control-`, `Alt-` or `Meta-`modifiers.
+/// This doesn't apply to the named keys on the keyboard, like `F1`.
+/// The escape sequence also doesn't apply to defined unicode input, like
+/// cyrillic chars.
+///
+/// For the details on how to implement the mode handling correctly
+/// consult the xterm's [`implementation`] and the output of the
+/// xterm's provided perl [`script`]. An example of the output
+/// is present [`here`].
+///
+/// [`implementation`]: https://invisible-island.net/xterm/modified-keys.html
+/// [`script`]: https://github.com/ThomasDickey/xterm-snapshots/blob/master/vttests/modify-keys.pl
+/// [`here`]: https://github.com/alacritty/vte/blob/master/doc/modifyOtherKeys-example.txt
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ModifyOtherKeys {
+    /// Reset the state.
+    Reset,
+    /// Enables this feature for keys except for those with well-
+    /// known behavior, e.g., Tab, Backarrow and some special
+    /// control character cases which are built into the X11
+    /// library, e.g., Control-Space to make a NUL, or Control-3
+    /// to make an Escape character.
+    ///
+    /// The following cases could be outlined based on the xterm
+    /// implementation when the escape sequences shouldn't be
+    /// created:
+    ///
+    /// - When the key is in range of `[0x40;0x7f]` and the modifier is either Control or Shift;
+    /// - when the key combination is a known control combination alias.
+    ///
+    /// For more details, consult the [`example`] of the suggested translation.
+    ///
+    /// [`example`]: https://github.com/alacritty/vte/blob/master/doc/modifyOtherKeys-example.txt
```

Comment:
I don't think there's any way to link to the filesystem here, is there? This link seems kinda easy to break.

Reply:
yeah, there's no way to do so. We could write an alternative markdown document and include it, but I think it's for the entire enum and it'll be very verbose.